connectors/netty-connector/src/main/java/org/glassfish/jersey/netty/connector/NettyConnector.java

```diff
@@ -202,6 +203,11 @@ protected void initChannel(SocketChannel ch) throws Exception {
                      p.addLast(new ChunkedWriteHandler());
                      p.addLast(new HttpContentDecompressor());
                      p.addLast(new JerseyClientHandler(NettyConnector.this, jerseyRequest, jerseyCallback, settableFuture));
+                     Integer readTimeout = ClientProperties.getValue(jerseyRequest.getConfiguration().getProperties(),
+                                                        ClientProperties.READ_TIMEOUT, 0);
+                     if (readTimeout > 0) {
+                          p.addFirst(new ReadTimeoutHandler(readTimeout, TimeUnit.MILLISECONDS));
```

Comment:
Would it make more sense to use https://netty.io/4.0/api/io/netty/handler/timeout/IdleStateHandler.html instead ?